Examples/Algorithms/Utilities/include/ActsExamples/Utilities/MeasurementMapSelector.hpp

```diff
@@ -0,0 +1,95 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "ActsExamples/EventData/IndexSourceLink.hpp"
+#include "ActsExamples/EventData/Measurement.hpp"
+#include "ActsExamples/Framework/DataHandle.hpp"
+#include "ActsExamples/Framework/IAlgorithm.hpp"
+#include "ActsExamples/Framework/WhiteBoard.hpp"
+#include "ActsFatras/EventData/Barcode.hpp"
+
+#include <string>
+#include <vector>
+
+namespace ActsExamples {
+
+/// Simple algorithm, that allows to extract a subset of a
+/// measurment-particles-map.
+/// This allows to conveniently work on subsets of the geometry.
+///
+class MeasurementMapSelector final : public IAlgorithm {
+  using Map = IndexMultimap<ActsFatras::Barcode>;
+
+ public:
+  struct Config {
+    /// Input spacepoints collection.
+    std::string inputMeasurementParticleMap;
+
+    /// Input source links
+    std::string inputSourceLinks;
+
+    /// Output protoTracks collection.
+    std::string outputMeasurementParticleMap;
+
+    /// What spacepoints to keep
+    std::vector<Acts::GeometryIdentifier> geometrySelection;
+  };
+
+  /// Constructor of the track finding algorithm
+  ///
+  /// @param cfg is the config struct to configure the algorithm
+  /// @param level is the logging level
+  MeasurementMapSelector(Config cfg, Acts::Logging::Level lvl)
+      : IAlgorithm("MeasurementMapSelector", lvl), m_cfg(cfg) {
+    m_inputSourceLinks.initialize(m_cfg.inputSourceLinks);
+    m_inputMap.initialize(m_cfg.inputMeasurementParticleMap);
+    m_outputMap.initialize(m_cfg.outputMeasurementParticleMap);
+  }
+
+  virtual ~MeasurementMapSelector() {}
```

Comment:
```suggestion
  virtual ~MeasurementMapSelector() = default;
```

Reply:
Hmm I agree that would be cleaner, but if the CI runs through I would keep it as it is I guess

Reply:
clang tidy may be against it :smile:

Reply:
Okay, but then I need a new review :laughing: